src/SectorView.cpp

```diff
@@ -585,48 +586,60 @@ std::vector<SystemPath> SectorView::GetRoute()
 	return m_route;
 }
 
-std::vector<SystemPath> SectorView::AutoRoute(const SystemPath &start, const SystemPath &target)
+void SectorView::AutoRoute(std::vector<SystemPath> &outRoute, const SystemPath &start, const SystemPath &target) const
```

Comment:
I don't think that returning the path by value caused the vector copy. In C++11 it surely will move it rather than copy, but even before that, the RVO would probably kick in. And even if it didn't, it's rather a small vector, so I'd expect no visible slowdown because of that.
But if you'd still like to keep it as an out param, I'd recommend to move it to the last param (it's just more popular way, it all style guides that I saw that's the recommendation), and to convert from reference to a pointer (that way the calling function will know that that parameter is probably modified).

Reply:
It was just my preference, and like a few of the others after this it's because I don't trust the compiler to actually do that. Sure sometimes it will use move instead of copy but not always, whereas this way it will always work 👍

Reply:
If being explicit and clear in the code doesn't add any downside, then I'm all for it, as at least I am very foggy on what magic goes on in the compiler.